handler/push.go

```diff
@@ -180,19 +180,21 @@ func splitLabels(labels string) (map[string]string, error) {
 	for i := 0; i < len(components)-1; i += 2 {
 		name, value := components[i], components[i+1]
 		trimmedName := strings.TrimSuffix(name, Base64Suffix)
+		unescapedName := model.UnescapeName(trimmedName, model.ValueEncodingEscaping)
 		if !model.LabelNameRE.MatchString(trimmedName) ||
+			!model.LabelName(unescapedName).IsValid() ||
```

Comment:
With this validity test, wo don't need the one in the previous line anymore, do wo?